### PR TITLE
distsql: move SelectResult to file "select_result.go"

### PR DESCRIPTION
**Pull request by zz-jason**
## What have you changed? (mandatory)

1. Move `SelectResult` from file "distsql/distsql.go" to file "distsql/select_result.go"
2. Remove `errInvalidResp` since it is not used.
3. Remove `readRowFromChunk()` since it is not used as well.
4. Add benchmark test: "BenchmarkDecodeToChunk" and "BenchmarkReadRowsData"

## What is the type of the changes? (mandatory)

- Improvement

## How has this PR been tested? (mandatory)

unit test

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

No

## Does this PR affect tidb-ansible update? (mandatory)
No

## Does this PR need to be added to the release notes? (mandatory)
No

### CODE REVIEW

**Comment by zz-jason** [comment]
@coocood @lysu PTAL

**Comment by shenli** [comment]
LGTM

File: distsql/distsql_test.go [hunk 1]
```diff
@@ -205,3 +209,78 @@ func (r *mockResultSubset) GetData() []byte { return r.data }
...
+	for i := 0; i < numCols; i++ {
+		colTypes[i] = &types.FieldType{Tp: mysql.TypeLonglong}
+	}
+	chk := chunk.NewChunkWithCapacity(colTypes, numRows)
+
+	buffer := populateBuffer()
+
+	b.ResetTimer()
+	for i := 0; i < b.N; i++ {
+		mockReadRowsData(buffer, colTypes, chk)
```
Comment (coocood) [inline distsql/distsql_test.go]: Should reset the chunk?

File: distsql/select_result.go [hunk 2]
```diff
@@ -0,0 +1,199 @@
+// Copyright 2017 PingCAP, Inc.
```
Comment (zimulala) [inline distsql/select_result.go]: s/2017/2018

File: distsql/distsql_test.go [hunk 3]
```diff
@@ -205,3 +209,79 @@ func (r *mockResultSubset) GetData() []byte { return r.data }
 
 // GetStartKey implements kv.Response interface.
 func (r *mockResultSubset) GetStartKey() kv.Key { return nil }
+
```
Comment (zimulala) [inline distsql/distsql_test.go]: Remove this blank line.

**Comment by zz-jason** [comment]
@coocood @zimulala PTAL

**Comment by coocood** [comment]
LGTM

**Pull request merged**